internal/cli/atlas/networking/peering/watch.go

```diff
@@ -56,10 +57,28 @@ func (opts *WatchOpts) watcher() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	if result.Status != "" {
-		return result.Status == waitingForUser || result.Status == failed || result.Status == available, nil
+
+	switch v := result.(type) {
+	case *atlasv2.AWSPeerVpc:
+		return watcherAWS(v), nil
+	case *atlasv2.AzurePeerNetwork:
+		return watcherAzure(v), nil
+	case *atlasv2.GCPPeerVpc:
+		return watcherGCP(v), nil
 	}
-	return result.StatusName == pendingAcceptance || result.StatusName == failed || result.StatusName == available, nil
+	return false, nil
+}
+
+func watcherGCP(peer *atlasv2.GCPPeerVpc) bool {
+	return *peer.Status == waitingForUser || *peer.Status == failed || *peer.Status == available
```

Comment:
given these are pointers I believe we generate getters if you have a `GetStatus()` method you can simplify this a lot 
```go
type StatusGetter interface {
  GetStatus()(something)
}

func watcher(peer StatusGetter) bool {
  return peer.GetStatus() == waitingForUser || peer..GetStatus() == failed || peer..GetStatus() == available
}
```
by using an interface you can make it a single method

Reply:
Great suggestion! The only problem for this is that the AWS method is called `GetStatusName()`, and the GCP and Azure ones are called `GetStatus()` - and I don't know if there is any Golang magic that we could use in this case